### PR TITLE
wordy 1.4.0

### PR DESCRIPTION
**Pull request by petertseng**

### CODE REVIEW

File: exercises/wordy/example.rs [hunk 1]
```diff
@@ -1,65 +1,56 @@
...
+    let words = c.trim_end_matches('?').split_whitespace().map(|word| {
+        if let Ok(i) = word.parse::<i32>() {
+            Token::Number(i)
+        } else {
+            Token::NonNumber(word)
         }
-        result = evaluate(result, opr, operand(&t.remove(0)));
-        Some(result)
+    }).collect::<Vec<_>>();
+    if words.len() < 3 {
```
Comment (petertseng) [inline exercises/wordy/example.rs]: there is no test case that exercises this condition. I am considering `What is?` as wordy 1.5.0

File: exercises/wordy/example.rs [hunk 2]
```diff
@@ -1,65 +1,56 @@
...
+        [
+            Token::NonNumber("What"),
+            Token::NonNumber("is"),
+            Token::Number(i),
+        ] => i,
+        _ => return None,
+    };
+    let mut words = words.split_at(3).1;
+    while !words.is_empty() {
+        let tmp = apply_op(result, words)?;
```
Comment (petertseng) [inline exercises/wordy/example.rs]: I would have liked to write `(result, words) = apply_op(result, words)?;` or something, but that is not valid. A shame.

File: exercises/wordy/example.rs [hunk 3]
```diff
@@ -1,65 +1,56 @@
...
-    let mut result: i32 = 0;
-    let mut opr = "plus";
-
-    if t.len() <= 1 {
-        None
-    } else {
-        while t.len() > 1 {
-            result = evaluate(result, opr, operand(&t.remove(0)));
-            opr = operator(&t.remove(0));
+    if !c.ends_with('?') {
```
Comment (petertseng) [inline exercises/wordy/example.rs]: no test case tests this line of code

File: exercises/wordy/example.rs [hunk 4]
```diff
@@ -1,65 +1,56 @@
...
+        return None;
+    }
+    let words = c.trim_end_matches('?').split_whitespace().map(|word| {
```
Comment (petertseng) [inline exercises/wordy/example.rs]: note that if there were a test case along the lines of `What is 1 + 2???` then that might raise questions as to whether that case is acceptable. This code would accept it.

File: exercises/wordy/example.rs [hunk 5]
```diff
@@ -1,65 +1,56 @@
...
+        if let Ok(i) = word.parse::<i32>() {
+            Token::Number(i)
+        } else {
+            Token::NonNumber(word)
         }
-        result = evaluate(result, opr, operand(&t.remove(0)));
-        Some(result)
+    }).collect::<Vec<_>>();
+    if words.len() < 3 {
+        return None;
```
Comment (petertseng) [inline exercises/wordy/example.rs]: I wish I didn't have to do this separately, but `words[0..3]` will panic if `words.len() < 3`. Maybe something with `iter().take(3)`, but then I hae to `collect` as well and it doesn't seem worth it.

File: exercises/wordy/example.rs (hunk 3 above)
Comment (coriolinus) [reply exercises/wordy/example.rs]: I see no reason to keep it, then. This implements a parser stricter than the tests and readme require.

File: exercises/wordy/example.rs (hunk 2 above)
Comment (petertseng) [reply exercises/wordy/example.rs]: that was pointed out in https://github.com/exercism/rust/pull/665#discussion_r222796492 . I wonder if I should also consider using that function that takes `&mut`. Note though that there still remains the possibility that the function will cause an early `return None` (currently done by the `?`), so there's a bit more to it

File: exercises/wordy/example.rs (hunk 5 above)
Comment (coriolinus) [reply exercises/wordy/example.rs]: I think `words[..3]` might not panic, being an unbounded range. Might be worth looking into, at least.

File: exercises/wordy/example.rs (hunk 2 above)
Comment (coriolinus) [reply exercises/wordy/example.rs]: Yeah, destructuring assignment would be a really nice usability improvement for the language.

File: exercises/wordy/example.rs (hunk 5 above)
Comment (petertseng) [reply exercises/wordy/example.rs]: Interesting. Looks like it still does though. `thread 'reject_empty' panicked at 'index 3 out of range for slice of length 0', libcore/slice/mod.rs:1932:5`

**Pull request merged**